pkgs/leak_tracker_flutter_testing/lib/src/leak_tracking_for_tests.dart

```diff
@@ -0,0 +1,183 @@
+    }
+
+    return settings.copyWith(
+      leakSkipLists: LeakSkipLists(
+        notGCed: settings.leakSkipLists.notGCed.merge(
+          LeakSkipList(
+            byClass: addClassesToMap(notGCed, classes),
+            skipAll: allNotGCed,
+          ),
+        ),
+        notDisposed: settings.leakSkipLists.notGCed.merge(
+          LeakSkipList(
+            byClass: addClassesToMap(notDisposed, classes),
+            skipAll: allNotDisposed,
+          ),
+        ),
+      ),
+    );
+  }
+
+  /// Returns copy of [settings] with reduced skip lists.
+  ///
+  /// Items in [classes] will be removed from all skip lists.
+  static LeakTrackingForTestsSettings withTracked({
+    List<String> notGCed = const [],
+    List<String> notDisposed = const [],
+    List<String> classes = const [],
+  }) {
+    final result = settings.copyWith(
+      leakSkipLists: LeakSkipLists(
+        notGCed: settings.leakSkipLists.notGCed.track(notGCed).track(classes),
+        notDisposed: settings.leakSkipLists.notDisposed
+            .track(notDisposed)
+            .track(classes),
+      ),
+    );
+    return result;
+  }
+
+  /// Removes certain classes and leak types from skip lists in [settings].
+  ///
+  /// Items in [classes] will be removed from all skip lists.
+  static void track({
+    List<String> notGCed = const [],
+    List<String> notDisposed = const [],
+    List<String> classes = const [],
+  }) {
+    settings = withTracked(
+      notDisposed: notDisposed,
+      notGCed: notGCed,
+      classes: classes,
+    );
+  }
+}
+
+/// Leak tracking settings for tests.
+///
+/// Should be instantiated using static methods of [LeakTrackingForTests].
+class LeakTrackingForTestsSettings {
+  LeakTrackingForTestsSettings._({
+    this.paused = true,
+    this.leakSkipLists = const LeakSkipLists(),
+    this.leakDiagnosticConfig = const LeakDiagnosticConfig(),
+    this.failOnLeaks = true,
+    this.onLeaks = _emptyLeakHandler,
+  });
+
+  /// Creates a copy of this object with the given fields replaced
+  /// with the new values.
+  LeakTrackingForTestsSettings copyWith({
+    LeakSkipLists? leakSkipLists,
+    LeakDiagnosticConfig? leakDiagnosticConfig,
+    bool? failOnLeaks,
+    LeaksCallback? onLeaks,
+    MemoryBaselining? baselining,
+    bool? paused,
+  }) {
+    return LeakTrackingForTestsSettings._(
+      leakSkipLists: leakSkipLists ?? this.leakSkipLists,
+      leakDiagnosticConfig: leakDiagnosticConfig ?? this.leakDiagnosticConfig,
+      failOnLeaks: failOnLeaks ?? this.failOnLeaks,
+      onLeaks: onLeaks ?? this.onLeaks,
+      paused: paused ?? this.paused,
+    );
+  }
+
+  /// If true, leak tracking is paused.
+  final bool paused;
+
+  /// If true, tests will fail on leaks.
+  ///
+  /// Set to true to test that tests collect expected leaks.
+  final bool failOnLeaks;
+
+  /// Callback to invoke if test collected leaks, before failing if [failOnLeaks] is true.
```

Comment:
wording nit: Callback to invoke before the test fails when [failOnLeaks] is true and leaks were found.